src/redisearch_rs/trie_bencher/src/corpus.rs

```diff
@@ -1,75 +1,200 @@
+/// must be further converted to the "legacy" string types: see [crate::strvec2_raw_words].
+///
+/// Information for remote files can be found in the folder: git_root/tests/benchmarks/
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum CorpusType {
+    /// uses a corpus from the redis benchmarks that contains 1k rows in a csv file describing wikipedia articles as a line
+    RedisBench1kWiki,
+
+    /// uses a corpus from the redis benchmarks that contains 10k rows in a csv file describing wikipedia articles as a line
+    RedisBench10kNumerics,
+
+    /// legacy corpus from gutenberg.net called 1984.txt
+    GutenbergEbook,
+}
+
+impl CorpusType {
+    /// If the corpus has already been downloaded, read it from disk.
+    /// Otherwise, download it from the internet and save it to disk.
+    /// In any case perform a crc32 check to notice changes in the corpus data.
+    pub fn download_or_read_corpus(&self) -> String {
+        let path = self.get_cached_path();
+        let corpus = if std::fs::exists(&path).ok() != Some(true) {
+            let corpus = download_corpus(self.get_url());
+            // ensure data folder exists
+            let cache_folder = PathBuf::from("data");
+            if !std::fs::exists(cache_folder)
+                .expect("Cannot check for existience of data folder (cache), check permissions")
+            {
+                std::fs::create_dir(PathBuf::from("data"))
+                    .expect("Failed to create data folder (cache)");
+            }
+            fs_err::write(&path, corpus.as_bytes()).expect("Failed to write corpus to disk");
+            corpus
+        } else {
+            fs_err::read_to_string(&path).expect("Failed to read corpus")
+        };
+
+        // check that the corpus hasn't been altered.
+        if let Some(stored_checksum) = self.get_checksum() {
+            let checksum = crc32fast::hash(corpus.as_bytes());
+            assert_eq!(
+                checksum, stored_checksum,
+                "The checksum of the downloaded corpus does not match the expected value. \
+                This may impact the accuracy and relevance of the operations being benchmarked. \
+                Review the diff before updating the expected checksum."
+            );
+        }
         corpus
-    } else {
-        fs_err::read_to_string(&corpus_path).expect("Failed to read corpus")
-    };
-
-    // Build a smaller corpus for microbenchmarks, limited to <100 unique words.
-    let bench_words = {
-        let mut unique = BTreeSet::new();
-        'outer: for line in corpus.lines().skip(36) {
-            for word in line.split_whitespace() {
-                if unique.insert(word.to_string()) && unique.len() > 82 {
-                    break 'outer;
+    }
+
+    /// Creates a vector of keys for insertion into a trie,
+    /// may output the trie to a file using a pretty printer.
```

Comment:
> may output the trie to a file using a pretty printer.

Why would one want that? Please add a sentence on that either here or at the call sites